aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -31,6 +31,98 @@
 namespace at {
 namespace native {
 
+enum class TypePromotionStrategy {
```

Comment:
In general, functions that are only used statically (within one file) should be declared + defined within an anonymous namespace, so they don't pollute the global at::native namespace.
```
namespace {
  promoteType1() {
  }
  promoteType2() {
  }
  ....
} // end anonymous namespace

unary_op_impl() {
  promoteType1(...)
}
```

Reply:
Thanks, @mcarilli. I've now committed changes using anonymous namespace!